pkg/cnsmigration/migrator.go

```diff
@@ -323,6 +342,20 @@ func (c *CNSVolumeMigrator) loginToVCenter(ctx context.Context) error {
 	if err = c.vSphereConnection.LoginToCNS(ctx); err != nil {
 		return fmt.Errorf("error logging into CNS: %v", err)
 	}
+	return nil
+}
 
+func (c *CNSVolumeMigrator) checkRequiredVersion() error {
+	apiVersion, build, err := c.vSphereConnection.GetVersionInfo()
+	if err != nil {
+		return fmt.Errorf("error getting vCenter version: %v", err)
+	}
+	match, message, err := utils.CheckForMinimumPatchedVersion(minRequirement, apiVersion, build)
+	if err != nil {
+		return fmt.Errorf("error checking for minimum version: %v", err)
+	}
+	if !match {
+		return fmt.Errorf("vCenter version %s:%s is not supported: minimum supported version is - %s", apiVersion, build, message)
+	}
```

Comment:
nit: can `CheckForMinimumPatchedVersion` return error when the version don't match? I don't think having a separate `match bool` is very useful. `match=false` has the same consequence as `err = nil` - both return error to the caller.

Reply:
So I have enhanced some of the code to return and print actual required version if parsing etc also fails. IMO - this will lead to better error reporting than the alternative.